### PR TITLE
Delete retry function

### PR DESCRIPTION
**Pull request by edify42**
Adds retry behaviour and writes a new record to the DynamoDB table with the `deleteCount` incremented
Modify the serverless.yml file to better support multiple stackjanitor deployments in 1 AWS account

### CODE REVIEW

File: serverless.yml [hunk 1]
```diff
@@ -48,6 +49,7 @@ provider:
       Resource: "*"
     - Effect: "Allow"
       Action:
+        - "elasticache:Delete*"
```
Comment (luizfelipepg) [inline serverless.yml:52]: Out of curiosity, is that any reason why we don't use only delete* for all resources we have here ?

Comment (luizfelipepg) [reply serverless.yml:52]: I know delete is the worst and won't make big difference but just asking out of curiosity :P

Comment (edify42) [reply serverless.yml:52]: I'm confused :thinking:  do you mean `elasticache:*` or ower case `elasticache:delete*`?

Comment (luizfelipepg) [reply serverless.yml:52]: I mean, looks like for all other we use `whatever:*` but for this one we using `elasticache:delete*`. I was just wondering why we don't use `whatever:delete*` as you did for elasticcache. Is there anything that stackjanitor does apart from delete things ? Maybe also scale down in some cases ?

Comment (edify42) [reply serverless.yml:52]: Oh yeah - it doesn't need to be this permissive that's true. I'm not too sure why but I'm sure we can eventually lock down the permissions further.
`ec2:*` is definitely concerning haha

File: src/handlers/deleteCloudFormationStack.ts [hunk 2]
```diff
@@ -73,6 +79,25 @@ async function processRecords(records: ParsedRecord<DataItem>[]) {
...
+        record.oldData?.deleteCount &&
+        record.oldData.deleteCount > config.MAX_CLEANUP_RETRY
+      ) {
+        // Log message to cloudwatch
+        logger.error(
+          `Failed to delete stack after ${config.MAX_CLEANUP_RETRY} additional attempts: ${oldData.stackName}`
+        );
+      } else {
+        // Recreate record
+        const deleteItem = generateRepeatedDeleteItem(oldData);
```
Comment (luizfelipepg) [inline src/handlers/deleteCloudFormationStack.ts:97]: So you adding it back to dynamodb to try again ?

File: serverless.yml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 service:
-  name: StackJanitor
+  name: StackJanitor${self:custom.NAME_SUFFIX}
```
Comment (luizfelipepg) [inline serverless.yml:2]: I just got a bit confuse why we need the name_suffix here. Is that for your testing ?

File: src/handlers/deleteCloudFormationStack.ts (hunk 2 above)
Comment (edify42) [reply src/handlers/deleteCloudFormationStack.ts:97]: Yeah exactly - the TTL will expire/auto delete the record and we need to add it back. Similarly the git webhook deletes the record and then this delete function attempts the CFn delete

File: serverless.yml (hunk 3 above)
Comment (edify42) [reply serverless.yml:2]: Yep - just so I could deploy multiple stack janitors in one account (they'd have the same CFn stack name without this suffix). Happy to keep it here as it's useful for future testing

**Comment by luizfelipepg** [comment]
LGTM 👍

**Review by luizfelipepg — APPROVED** [review]
LGTM 👍

**Review by lexko — APPROVED** [review]

**Pull request merged**